test/test_parser.py

```diff
@@ -0,0 +1,82 @@
+import os
+import json
+from typing import Any
+
+from pyfakefs import fake_filesystem_unittest
+
+import cabinets
+from cabinets import Parser
+
+
+class MockTextParser(Parser):
+
+    @classmethod
+    def dump_content(cls, data: Any, **kwargs):
+        data['mock'] = 'parser'
+        return json.dumps(data)
+
+    @classmethod
+    def load_content(cls, content: bytes, **kwargs):
+        return {'mock-parser': content.decode('utf-8')}
+
+
+class TestParserArgument(fake_filesystem_unittest.TestCase):
+    fixture_path = os.path.join(os.path.dirname(__file__), 'fixtures')
+
+    def setUp(self):
+        self.setUpPyfakefs()
+        self.fs.add_real_directory(self.fixture_path)
+
+    def test_read_plain_text_default_parser(self):
+        protocol = 'file'
+        filename = os.path.join(self.fixture_path, 'sample_small.txt')
+        data = cabinets.read(f'{protocol}://{filename}', parser=True)
+        expected = "I am sample text!"
+        self.assertIsInstance(data, str)
+        self.assertEqual(expected, data)
+
+    def test_read_plain_text_no_parser(self):
+        protocol = 'file'
+        filename = os.path.join(self.fixture_path, 'sample_small.txt')
+        data = cabinets.read(f'{protocol}://{filename}', parser=False)
+        expected = bytes("I am sample text!", encoding='utf-8')
+        self.assertIsInstance(data, bytes)
+        self.assertEqual(expected, data)
+
+    def test_read_plain_text_custom_parser(self):
+        protocol = 'file'
+        filename = os.path.join(self.fixture_path, 'sample_small.txt')
+        data = cabinets.read(f'{protocol}://{filename}', parser=MockTextParser)
+        expected = {'mock-parser': "I am sample text!"}
+        self.assertIsInstance(data, dict)
+        self.assertDictEqual(expected, data)
+
+    def test_create_json_default_parser(self):
+        protocol, filename = 'file', 'tmp/sample.json'
+        content = {"hello": "world"}
+        cabinets.create(f'{protocol}://{filename}', content, parser=True)
+        with open(filename) as fh:
+            data = fh.read()
+        self.assertEqual('{"hello": "world"}', data)
+
+    def test_create_json_default_parser_fails(self):
+        protocol, filename = 'file', 'tmp/sample.json'
+        content = bytes('{"test":"bytes"}', encoding="utf-8")
```

Comment:
The lack of spacing makes me think we should pass `kwargs` to the `json.dumps()` and `json.loads()` methods so we can add arguments for indentation or different delimiters and whatnot.

Reply:
Yes. We pass `kwargs` already, but I agree we should pull out specific `kwargs` within the different parsers. Definitely useful for `csv` as well.

Reply:
Right so I guess I'm asking if you wanna do that [here](https://github.com/lucasmlofaro/cabinets/blob/main/cabinets/parser/json_parser.py#L11) in this PR or if you think we should do more of what you're talking about in a separate PR?

Reply:
I think a separate PR is a better idea at this point. Will be easier to review.